DESIGN.md

```diff
@@ -187,3 +187,62 @@ the producers/consumers adding records for non-replicated and replicated
 entries. Since each entry would have a sequence number, calculating the
 difference between the sequence numbers of the latest non-replicated and
 replicated records would give us the required statistics.
+
+## WRITING EXTENSIONS
+
+A backbeat extension allows to add more functionality to the core
+backbeat asynchronous processor. E.g. Asynchronous Replication is one
+of the extensions available for backbeat.
+
+Extensions are located in the extensions/ directory, with a
+sub-directory named after the extension name (lowercase).
+
+### Extending the Queue Populator
+
+The queue populator is a core backbeat process which reads entries
+from metadata log periodically and provides them to all running
```

Comment:
~from metadata log~ from a metadata log

Reply:
will say `from the metadata log` since normally there's only one log to read from (or none).